pycytominer/cyto_utils/sqlite.py

```diff
@@ -0,0 +1,561 @@
+        for col in columns:
+            # the sql below seeks to efficiently detect existence of values which
+            # do not match the column affinity type (for ex. a string in an integer column).
+            sql_stmt = """
+            SELECT
+                EXISTS(
+                    SELECT 1 FROM {table_name} 
+                    WHERE TYPEOF({col_name}) NOT IN ({col_types})
+                );
+            """
+            # join formatted string for use with sql query in {col_types} var
+            col_types = ",".join(
+                ["'{}'".format(x.lower()) for x in SQLITE_AFF_REF[col[2]]]
+            )
+
+            # there are challenges with using sqlalchemy vars in the same manner as above
+            # so we use format here along with nosec
+            result = connection.execute(
+                sql_stmt.format(
+                    table_name=col[0],
+                    col_name=col[1],
+                    col_types=col_types,
+                )
+            ).fetchall()[0][
+                0
+            ]  # nosec
+            if result > 0:
+                # if our result is greater than 0 it means values with conflicting storage
+                # class existed within the focus column and as a result, we return False
+                logger.warning(
+                    "Discovered conflicting %s column %s affinity type and storage class.",
+                    col[0],
+                    col[1],
+                )
+                return True
+
+    # return false if we did not find conflicting affinity vs storage class values
+    logger.info(
+        "Found no conflicting affinity vs storage class data within provided database."
+    )
+
+    return False
+
+
+def contains_str_like_null(
+    sql_engine: Union[str, Engine],
+    table_name: Optional[str] = None,
+    column_name: Optional[str] = None,
+) -> bool:
+    """
+    Detect whether the given database, table, or column contains
+    a string value which is similar to NULL. Strings instead of
+    SQLite NULL may be interpreted at read as a string (value)
+    instead of a NULL (non-value).
+
+    Parameters
+    ----------
+    sql_engine: str | sqlalchemy.engine.base.Engine
+        filename of the SQLite database or existing sqlalchemy engine
+    table_name: str
+        optional specific table name to check within database
+    column_name: str
+        optional specific column name to check within database
+
+    Returns
+    -------
+    bool
+        Returns True if found a str value similar to null, else returns False.
+    """
+
+    logger.info("Determining if SQLite database contains string values like NULL's.")
+
+    # create an engine
+    engine = engine_from_str(sql_engine)
+
+    # gather columns to be used below
+    columns = collect_columns(engine, table_name, column_name)
+
+    # strings which are like nulls for later use in below SQL 'in'
+    like_nulls_str_list = ",".join(["'{}'".format(x) for x in LIKE_NULLS])
```

Comment:
I like the idea of passing to a function as you mentioned to keep things flexible and more traceable. I'll make a change to this effect. The upfront global was partially an effort to make it really visible and more easily maintained - it feels like an area up for debate more generally so I didn't want it hidden (for ex: I originally had `'na'` as an additional LIKE_NULLS value but removed it because it felt like an overreach). Non-value's get challenging when we include multiple languages and libraries! 🙂 

Would it be beneficial to explore the use of [`os.environ`](https://docs.python.org/3/library/os.html#os.environ) for this in addition to the function arg?